mesonbuild/build.py

```diff
@@ -1062,8 +1062,17 @@ def add_deps(self, deps):
     def get_external_deps(self):
         return self.external_deps
 
+    def is_internal(self):
+        if isinstance(self, StaticLibrary) and not self.need_install:
+            return True
+        return False
+
     def link(self, target):
         for t in listify(target, unholder=True):
+            if isinstance(self, StaticLibrary) and t.is_internal():
```

Comment:
Wont this result in convenience libraries linking in other convenience libraries instead of linking both into the final object:

say we have C1, and C2 as convenience libraries and T is an installed target of any type. My expectation would be that if C1 links C2 that C1 would *not* contain C1's libs, but instead any target linking C1 would also link C2; but that if T was a static lib it would include all of the objects from both C1 and C2, since it's being installed.

Reply:
Ah, to answer my own question, this if fixed in a later patch.